src/renderer/components/VideoPreview/VideoPreviewController.tsx

```diff
@@ -225,6 +233,29 @@ const VideoPreviewControllerBase = (
     );
   }, [currentProject?.mediaFilePath]);
 
+  // If the state of play/pause is changed, play/pause the video preview
+  // if the video is paused, update the time in redux
+  useEffect(() => {
+    if (playState) {
+      play();
+    } else {
+      pause();
+      store.dispatch(
+        videoSeek({
+          time: clockRef.current.time,
+          lastUpdated: new Date(),
+        } as UpdatedTimeSeek)
+      );
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [playState]);
+
+  // if the state of playback time is changed, set the time accordingly for video preview
+  useEffect(() => {
+    setPlaybackTime(timeState);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [timeState]);
```

Comment:
setPlaybackTime

Reply:
If I include the dependency, it creates an infinite loop.

Reply:
You can fix this by wrapping `setPlaybackTime` in a `useCallback`. Best to add all relevant dependencies to the useEffect because if you don't, it'll be calling an old version of the function which had a closure around some outdated state. Kind of hard to explain. General practice is just wrap functions in `useCallback` and variable assignments in `useMemo` so they don't change every re-render

Reply:
Let me know if you need help with this